eng/pipelines/templates/jobs/archetype-sdk-tests.yml

```diff
@@ -3,20 +3,47 @@ parameters:
   EnvVars: {}
   MaxParallel: 0
   Matrix:
-    Win2016:
+    Linux - Java 8:
+      OSName: 'Linux'
+      OSVmImage: 'ubuntu-16.04'
+      JavaVersion: '1.8'
+      DisplayName: '$(OSName) on Java $(JavaVersion)'
```

Comment:
Why do we need to pass this into all the matrix? You should be able to just have this exact string in the publish results to avoid this extra duplicated parameter.

Reply:
See https://github.com/Azure/azure-sdk-for-java/blob/master/eng/pipelines/templates/jobs/archetype-sdk-client.yml#L48-L51 which will give you an example of how to set DisplayName at runtime if it isn't set.

Reply:
Ignore my last comment as I was thinking of something else.

The snippet you want will look something like:
```
- pwsh: |
    echo "##vso[task.setvariable variable=DisplayName]$(OSName) on Java $(JavaVersion)"
  displayName: "Setup DisplayName"
  condition: eq(variables['DisplayName'], '')
```